test/functional/tests/conftest.py

```diff
@@ -64,15 +64,16 @@ def prepare_and_cleanup(request):
                 IP(dut_config['ip'])
             except ValueError:
                 raise Exception("IP address from configuration file is in invalid format.")
-        dut_config = test_wrapper.prepare(request.param, dut_config)
+        dut_config = test_wrapper.prepare(dut_config)
 
-    TestRun.prepare(dut_config)
+    TestRun.setup(dut_config)
 
     TestRun.plugins['opencas'] = {'already_updated': False}
 
-    TestRun.LOGGER.info(f"**********Test {request.node.name} started!**********")
-    yield
+    TestRun.LOGGER.info(f"**********Test {item.name} started!**********")
+
 
+def pytest_runtest_teardown(item, nextitem):
```

Comment:
I assume item and nextitem are required for this to be a python hook? Because I don't like unused variables :D

Reply:
Yes, they are.